tests/utils/log/test_log_reader.py

```diff
@@ -79,29 +81,66 @@ def configure_loggers(self, log_dir, settings_folder):
 
     @pytest.fixture(autouse=True)
     def prepare_log_files(self, log_dir):
-        dir_path = f"{log_dir}/{self.DAG_ID}/{self.TASK_ID}/2017-09-01T00.00.00+00.00/"
-        os.makedirs(dir_path)
-        for try_number in range(1, 4):
-            with open(f"{dir_path}/{try_number}.log", "w+") as f:
-                f.write(f"try_number={try_number}.\n")
-                f.flush()
+        for dag_id in {self.DAG_ID, self.DUMMY_DAG_ID, self.EXTERNAL_TASK_MARKER_DAG_ID}:
+            dir_path = f"{log_dir}/{dag_id}/{self.TASK_ID}/2017-09-01T00.00.00+00.00/"
+            os.makedirs(dir_path)
+            for try_number in range(1, 4):
+                with open(f"{dir_path}/{try_number}.log", "w+") as f:
+                    f.write(f"try_number={try_number}.\n")
+                    f.flush()
 
     @pytest.fixture(autouse=True)
-    def prepare_db(self, session, create_task_instance):
-        ti = create_task_instance(
+    def prepare_db(self, session, create_task_instance_of_operator):
+        from airflow.operators.bash import BashOperator
+
+        ti = create_task_instance_of_operator(
+            operator_class=BashOperator,
             dag_id=self.DAG_ID,
             task_id=self.TASK_ID,
             start_date=self.DEFAULT_DATE,
             run_type=DagRunType.SCHEDULED,
             execution_date=self.DEFAULT_DATE,
             state=TaskInstanceState.RUNNING,
+            bash_command="echo 'hello'",
         )
         ti.try_number = 3
         self.ti = ti
         yield
         clear_db_runs()
         clear_db_dags()
 
+    @pytest.fixture(scope="function")
+    def dummy_ti(self, session, create_task_instance):
+        dummy_ti = create_task_instance(
+            dag_id=self.DUMMY_DAG_ID,
+            task_id=self.TASK_ID,
+            start_date=self.DEFAULT_DATE,
+            run_type=DagRunType.SCHEDULED,
+            execution_date=self.DEFAULT_DATE,
+            state=TaskInstanceState.RUNNING,
+        )
+        dummy_ti.operator = 'DummyOperator'
+        self.dummy_ti = dummy_ti
+        yield
+        clear_db_runs()
+        clear_db_dags()
+
+    @pytest.fixture(scope="function")
+    def external_task_marker_ti(self, session, create_task_instance):
+        external_task_marker_ti = create_task_instance(
+            dag_id=self.EXTERNAL_TASK_MARKER_DAG_ID,
+            task_id=self.TASK_ID,
+            start_date=self.DEFAULT_DATE,
+            run_type=DagRunType.SCHEDULED,
+            execution_date=self.DEFAULT_DATE,
+            state=TaskInstanceState.RUNNING,
+        )
+        external_task_marker_ti.operator = 'ExternalTaskMarker'
+        self.external_task_marker_ti = external_task_marker_ti
+        yield
```

Comment:
You can instead `yield external_task_marker_ti` and use the fixture directly instead of `self.external_task_marker_ti` in tests. This makes the code a bit easier to trace IMO.